### PR TITLE
maintain maximized state when clicking dock icon

### LINKED ISSUE #22087: MacOS: App window shrinks when clicking Dock icon
### Steps To Reproduce

1. Open Bitwarden Desktop app
2. Maximize Bitwarden Desktop window (double click title bar, depends on Mac settings)
2a. Optionally, open a different app so the Bitwarden window is hidden (bug triggers either way)
3. Click Bitwarden icon in the Dock

### Expected Result

Nothing happens (or if the Bitwarden app is behind other windows, it comes to the front without changing size).

### Actual Result

Bitwarden window shrinks (regardless of being behind other windows or not).

### Screenshots or Videos

Triggered the bug a few times in the video below.

https://github.com/user-attachments/assets/aa9ddeaa-4f4e-447f-bccd-b9a8741dd070

### Additional Context

Problem does not occur with Bitwarden desktop v2026.6.1, so it is a regression in v2026.7.0.
Problem does not occur by switching windows a different way (e.g. mission control, Cmd+Tab cycle), it is clearly triggered by the click on the Bitwarden icon in the Dock.

Bitwarden info:
Version 2026.7.0
SDK 'main (3e060f8)'
Shell 41.7.2
Renderer 146.0.7680.216
Node 24.15.0
Architecture arm64

MacOS info:
MacOS M1 Pro
Tahoe 26.5.2 (25F84)

### Operating System

macOS

### Operating System Version

Tahoe 26.5.2 (25F84)

### Installation method

Direct Download (from bitwarden.com)

### Build Version

2026.7.0

### Issue Tracking Info

- [x] I understand that work is tracked outside of GitHub. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

### PR DESCRIPTION
**Pull request by addisonbeck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-41034
Fixes https://github.com/bitwarden/clients/issues/22087

## 📔 Objective

Fix a bug involving the logic that focuses the desktop app's window not respecting if the latest window size was maximized.

## 📸 Screenshots

First video is me recreating the issue against the latest prod artifacts. Second video is from this branch, showing the bug resolved. Ignore the fact that I have to blow up my desktop to click the dock, I have the dock disabled and did that as a workaround for testing.

https://github.com/user-attachments/assets/cf8b9c9f-224d-427e-a966-6860e11c5043

https://github.com/user-attachments/assets/1ef372d6-1bcb-452c-aa04-3535343dd685

### CODE REVIEW

**Review by coroiu — DISMISSED** [review]

File: apps/desktop/src/main/window.main.spec.ts [hunk 1]
```diff
@@ -106,4 +108,94 @@ describe("WindowMain", () => {
...
+      mockWin.isVisible.mockReturnValue(true);
+      sut.win = mockWin;
+      (sut as any).windowStates["mainWindowSize"] = { isMaximized: true };
+
+      // Act
+      sut.show();
+
+      // Assert
+      expect(mockWin.maximize).toHaveBeenCalledTimes(1);
+    });
```
[the remark this answers is not in the document]
Comment (addisonbeck) [reply apps/desktop/src/main/window.main.spec.ts:142]: Resolved on [set maximized state unconditionally in show()](https://github.com/bitwarden/clients/pull/22199/commits/9f3fb0e928ab903142e335aac637ea9ef0ac6812)

**Comment by addisonbeck** [comment]
Here's a screen recording of me running through different merthods for resizing, showing, and hiding the window. At the end I test minimizing with the menu bar and clicking the dock icon, which was an edge case caught by QA. The app restores as expected in all cases.

https://github.com/user-attachments/assets/951604f4-4313-4a2b-8014-5f9beb13fe47

**Review by coroiu — DISMISSED** [review]

**Comment by addisonbeck** [comment]
Sorry about the ping folks. I'm not sure what happened with this merge. I'm going to reopen this PR.
